index.js

```diff
@@ -3,29 +3,67 @@ import pMemoize from 'p-memoize';
 import delay from 'delay';
 
 const getDevice = pMemoize(async () => {
-	if (process.platform !== 'darwin') {
-		throw new Error('macOS only');
+	if (process.platform == 'darwin') {
+		const { stdout } = await execa('networksetup', ['-listallhardwareports']);
+		const result = /Hardware Port: Wi-Fi\nDevice: (?<device>en\d)/.exec(stdout);
+		if (!result) {
+			throw new Error('Couldn\'t find a Wi-Fi device');
+		}
+
+		return result.groups.device;
+	} else if (process.platform == 'win32') {
+		const { stdout } = await execa('powershell', [
+			'-command',
+			'Get-NetAdapter -Name Wi-Fi | Format-List -Property PnPDeviceID',
+		]);
+
+		const result = /(?<=PnPDeviceID : ).*/.exec(stdout);
+
+		if (!result) {
+			throw new Error('Couldn\'t find a Wi-Fi device');
+		}
+
+		return result[0];
+	} else {
+		throw new Error('macOS or Windows only');
 	}
-
-	const {stdout} = await execa('networksetup', ['-listallhardwareports']);
-
-	const result = /Hardware Port: Wi-Fi\nDevice: (?<device>en\d)/.exec(stdout);
-	if (!result) {
-		throw new Error('Couldn\'t find a Wi-Fi device');
-	}
-
-	return result.groups.device;
 });
 
-const isOn = async device => {
-	const {stdout} = await execa('ifconfig', [device]);
-	return stdout.includes('status: active');
+const isOn = async (device) => {
+	if (process.platform == 'darwin') {
+		const { stdout } = await execa('ifconfig', [device]);
+		return stdout.includes('status: active');
+	} else if (process.platform == 'win32') {
+		const { stdout } = await execa('powershell', [
```

Comment:
I think you can use the `shell` option for `execa`. You could always create a `execa` wrapper for Powershell, since it's used in multiple places.

Reply:
Great suggestion, I've implemented it on the latest commit.